src/server/helpers/file.py

```diff
@@ -45,3 +46,18 @@ def write_file(
     path = fs.get_path(fullpath)
 
     return path
+
+
+def create_file_filter(project: Project) -> Callable[[str], bool]:
+    fs = project.filesystem
+    fullpath = fs.get_fullpath(".gitignore")
+    if not fullpath.exists():
+        return lambda _: False
+
+    matches = parse_gitignore(fullpath)  # type: ignore
+
+    def ignore_path(path: str) -> bool:
+        file_exists: Callable[[str], bool] = lambda path: True if matches(path) else False
```

Comment:
@shashigharti 
Any reason you create a lambda here instead of just using the expression? =)

Reply:
it is due to to type error message.

Reply:
If you do like this?  
```
def file_filter(path: str) -> bool:
        return True if matches(path) else False
```